app/src/main/java/net/gsantner/markor/ui/hleditor/Highlighter.java

```diff
@@ -168,79 +168,24 @@ protected boolean isFirstHighlighting() {
     // Create spans
     //
 
-    /**
-     * Create Span for isMatching in ParcelableSpan's. Note that this will highlight the full matched pattern
-     * (including optionals) if no group parameters are given.
-     *
-     * @param spannable     Text spannable
-     * @param pattern       The pattern to match
-     * @param creator       A ParcelableSpanCreator for ParcelableSpan
-     * @param groupsToMatch (optional) groups to be matched, indexes start at 1.
-     */
-    protected static void createSpanForMatches(final Spannable spannable, final Pattern pattern, final SpanCreator.ParcelableSpanCreator creator, int... groupsToMatch) {
-        if (groupsToMatch == null || groupsToMatch.length < 1) {
-            groupsToMatch = new int[]{0};
-        }
-        int i = 0;
-        for (Matcher m = pattern.matcher(spannable); m.find(); i++) {
-            ParcelableSpan span = creator.create(m, i);
-            if (span != null) {
-                for (int g : groupsToMatch) {
-                    if (g == 0 || g <= m.groupCount()) {
-                        spannable.setSpan(span, m.start(g), m.end(g), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
-                    }
-                }
-            }
-        }
+    public interface SpanCreator<SpanType> {
+        SpanType create(final Matcher matcher, final int iM);
     }
 
-
-    /**
-     * Create Span for isMatching in paragraph's. Note that this will highlight the full matched pattern
-     * (including optionals) if no group parameters are given.
-     *
-     * @param spannable     Text spannable
-     * @param pattern       The pattern to match
-     * @param creator       A ParcelableSpanCreator for ParcelableSpan
-     * @param groupsToMatch (optional) groups to be matched, indexes start at 1.
-     */
-    protected static void createSpanForMatchesP(final Spannable spannable, final Pattern pattern, final SpanCreator.ParagraphStyleCreator creator, int... groupsToMatch) {
+    protected static <SpanType> void createSpanForMatches(final Spannable spannable, final Pattern pattern, final SpanCreator<SpanType> creator, int... groupsToMatch) {
```

Comment:
This change is unrelated to day / night colors per-se. It is just  a cleanup I been meaning to make.